geoh5py/workspace/workspace.py

```diff
@@ -678,50 +574,168 @@ def find_type(
-        """Get all active Object entities registered in the workspace.
+        else:  # Extract all objects uuid with matching name
+            list_entity_uid = [
+                key for key, val in self.list_entities_name.items() if val == name
+            ]
+
+        entity_list: List[Optional[Entity]] = []
+        for uid in list_entity_uid:
+            entity_list.append(self.find_entity(uid))
+
+        return entity_list
+
+    @property
+    def h5file(self) -> str:
         """
-        weakref_utils.remove_none_referents(self._objects)
-        return [cast("object_base.ObjectBase", v()) for v in self._objects.values()]
+        :str: Target *geoh5* file name with path
+        """
+        return self._h5file
 
-    def find_object(self, object_uid: uuid.UUID) -> Optional["object_base.ObjectBase"]:
+    @property
+    def list_data_name(self) -> Dict[uuid.UUID, str]:
         """
-        Find an existing and active Object
+        :obj:`dict` of UUID keys and name values for all registered Data
         """
-        return weakref_utils.get_clean_ref(self._objects, object_uid)
+        data_name = {}
+        for key, val in self._data.items():
+            entity = val.__call__()
+            if entity is not None:
+                data_name[key] = entity.name
+        return data_name
 
-    def all_data(self) -> List["data.Data"]:
-        """Get all active Data entities registered in the workspace.
+    @property
+    def list_entities_name(self) -> Dict[uuid.UUID, str]:
         """
-        weakref_utils.remove_none_referents(self._data)
-        return [cast("data.Data", v()) for v in self._data.values()]
+        :obj:`dict` of UUID keys and name values for all registered Entities
+        """
+        entities_name = self.list_groups_name
+        entities_name.update(self.list_objects_name)
+        entities_name.update(self.list_data_name)
+        return entities_name
 
-    def find_data(self, data_uid: uuid.UUID) -> Optional["data.Data"]:
+    @property
+    def list_groups_name(self) -> Dict[uuid.UUID, str]:
         """
-        Find an existing and active Data entity
+        :obj:`dict` of UUID keys and name values for all registered Groups
         """
-        return weakref_utils.get_clean_ref(self._data, data_uid)
+        groups_name = {}
+        for key, val in self._groups.items():
+            entity = val.__call__()
+            if entity is not None:
+                groups_name[key] = entity.name
+        return groups_name
 
-    def find_entity(self, entity_uid: uuid.UUID) -> Optional["Entity"]:
-        """Get all active entities registered in the workspace.
+    @property
+    def list_objects_name(self) -> Dict[uuid.UUID, str]:
         """
-        return (
-            self.find_group(entity_uid)
-            or self.find_data(entity_uid)
-            or self.find_object(entity_uid)
-        )
+        :obj:`dict` of UUID keys and name values for all registered Objects
```

Comment:
missing periods and :obj: reference
```suggestion
        :obj:`dict` of :obj:`uuid.UUID` keys and name values for all registered Objects.
```